ComponentKit/Components/CKNetworkImageComponent.mm

```diff
@@ -125,6 +125,12 @@ - (void)setSpecifier:(CKNetworkImageSpecifier *)specifier
   if (CKObjectIsEqual(specifier, _specifier)) {
     return;
   }
+  
+  BOOL isShowingCurrentImageURL = self.image != _specifier.defaultImage && self.image != nil;
+  if (isShowingCurrentImageURL && CKObjectIsEqual(_specifier.url, specifier.url)) {
+    _specifier = specifier;
+    return;
```

Comment:
Breaks if cropRect changes.

I'd rather we model this with an explicit state enum and state machine.